src/components/Camera.ts

```diff
@@ -0,0 +1,205 @@
+            return createElement("div", { className: "parent" },
+                createElement("img", {
+                    src: this.state.screenshot,
+                    style: this.createStyle(),
+                    alt: "Image path could not be found!"
+                }),
+                createElement("span", { className: "buttonContainer1", onClick: this.retakePicture },
+                this.createIcons(this.props.captureButtonName, "glyphicon glyphicon-" + this.props.captureButtonIcon)),
+                createElement("span", {
+                    className: "buttonContainer2",
+                    onClick: () => this.props.onClickAction({
+                        src: this.state.screenshot,
+                        id: this.state.pictureId
+                    })
+                }, this.createIcons(this.props.usePictureButtonName,
+                    "glyphicon glyphicon-" + this.props.usePictureButtonIcon)
+                )
+            );
+        }
+
+        return createElement("div", { className: "parent" },
+            createElement(WebCam, {
+                audio: false,
+                ref: this.setCameraReference,
+                screenshotFormat: "image/".concat(this.props.fileType),
+                style: this.createStyle()
+            }),
+            createElement("span", {
+                className: "buttonContainer",
+                onClick: this.takePicture
+            }, this.createIcons(this.props.captureButtonName, "glyphicon glyphicon-" + this.props.captureButtonIcon)),
+            this.createSwitchCameraButton()
+
+        );
+    }
+
+    componentWillMount() {
+        navigator.mediaDevices.enumerateDevices()
+            .then((devices: {}[]) => {
+                devices.forEach((device: { kind: string, deviceId: string }) => {
+                    if (device.kind === "videoinput") {
+                        this.availableDevices.push(device.deviceId);
+                    }
+                });
+                if (this.availableDevices.length >= 1) {
+                    this.setState({ cameraDevicePosition: this.availableDevices.length - 1 });
+                    this.getStream();
+                }
+            })
+            .catch((error: Error) => {
+                mx.ui.error(error.name + ": " + error.message);
+            });
+    }
+
+    componentDidUpdate() {
+        this.getStream();
+    }
+
+    private setCameraReference(webcam: Webcam) {
+        this.webcam = webcam;
+    }
+
+    private takePicture() {
+        this.setState({
+            screenshot: this.webcam.getScreenshot(),
+            pictureTaken: true,
+            pictureId: this.webcam.stream.id.concat("." + this.props.fileType)
+        });
+    }
+
+    private retakePicture() {
+        this.setState({ pictureTaken: false });
+    }
+
+    private changeCamera() {
+        if ((this.state.cameraDevicePosition) < (this.availableDevices.length - 1)) {
```

Comment:
Put the value in a const and set the state in one line.
`const cameraDevicePosition = this.state.cameraDevicePosition < (this.availableDevices.length - 1) ?
 this.state.cameraDevicePosition+1 : 0;`
`this.setState{cameraDevicePosition};`